peerdid/core/did_doc_types.py

```diff
@@ -0,0 +1,219 @@
+        if (
+            ver_method_type_str
+            == VerificationMethodTypeAgreement.JSON_WEB_KEY_2020.value
+            or ver_method_type_str
+            == VerificationMethodTypeAuthentication.JSON_WEB_KEY_2020.value
+        ):
+            if VerificationMethodField.JWK.value not in value:
+                raise MalformedPeerDIDDocError(
+                    "Invalid verification method: no {} field in {} method type".format(
+                        VerificationMethodField.JWK.value, ver_method_type_str
+                    )
+                )
+            return get_verification_method_type(
+                value[VerificationMethodField.JWK.value]
+            )
+
+        try:
+            return VerificationMethodTypeAgreement(ver_method_type_str)
+        except ValueError:
+            pass
+
+        try:
+            return VerificationMethodTypeAuthentication(ver_method_type_str)
+        except ValueError:
+            pass
+
+        raise MalformedPeerDIDDocError(
+            "Unknown verification method type {}".format(ver_method_type_str)
+        )
+
+    @classmethod
+    def from_dict(cls, value: dict):
+        if "id" not in value:
+            raise MalformedPeerDIDDocError("No 'id' field in method {}".format(value))
+        if "type" not in value:
+            raise MalformedPeerDIDDocError("No 'type' field in method {}".format(value))
+
+        ver_method_type = cls._get_ver_method_type(value)
+        format, field = cls._get_public_key_format(ver_method_type)
+        ver_material_cls = (
+            VerificationMaterialAgreement
+            if isinstance(ver_method_type, VerificationMethodTypeAgreement)
+            else VerificationMaterialAuthentication
+        )
+        ver_material = ver_material_cls(
+            value=value[field.value],
+            format=format,
+            type=ver_method_type,
+        )
+
+        return cls(
+            id=value["id"], controller=value["controller"], ver_material=ver_material
+        )
+
+
+class DIDCommServicePeerDID:
+    def __init__(
+        self,
+        id: str,
+        service_endpoint: Optional[str],
+        routing_keys: Optional[List[str]],
+        accept: Optional[List[str]],
+    ) -> None:
+        self.id = id
+        self.service_endpoint = service_endpoint
+        self.routing_keys = routing_keys
+        self.accept = accept
+
+    def to_dict(self):
+        res = {
+            SERVICE_ID: self.id,
+            SERVICE_TYPE: SERVICE_DIDCOMM_MESSAGING,
+            SERVICE_ENDPOINT: self.service_endpoint,
+            SERVICE_ACCEPT: self.accept,
+        }
+        if self.service_endpoint:
+            res[SERVICE_ENDPOINT] = self.service_endpoint
+        if self.routing_keys:
+            res[SERVICE_ROUTING_KEYS] = self.routing_keys
+        if self.accept:
+            res[SERVICE_ACCEPT] = self.accept
+        return res
+
+    @classmethod
+    def from_dict(cls, values: dict):
```

Comment:
return value type is not clear: I thought it should be an instance of `DIDCommServicePeerDID`

Reply:
The actual type is `List[Union[Dict, DIDCommServicePeerDID]]`. A service is not necessary of DIDCommService type, we have to support other (unknown) services (just as dicts).
I agree that List here is confusing, changed it so that `DIDCommServicePeerDID.from_dict` returns `Union[Dict, DIDCommServicePeerDID]` now.